### PR TITLE
[init_cfg.json.j2] remove hardcode in init_cfg.json.j2

### PR DESCRIPTION
**Pull request by stepanblyschak**
In order to avoid some missed container/feature listed in
init_cfg.json.j2 in FEATURE and CONTAINER_FEATURE tables I changed the
way optional feature are defined. With this change it is declared by
feature makefile whether this feature is optional or not. As for
CONTAINER_FEATURE table, the configuration is generated based on all
installable dockers.

Signed-off-by: Stepan Blyschak <stepanb@mellanox.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by liat-grozovik** [comment]
retest vsimage please

File: files/build_templates/init_cfg.json.j2 [hunk 1]
```diff
@@ -14,19 +16,18 @@
...
+        "{{ feature }}": {
             "status": "disabled"
         }{% if not loop.last %},{% endif -%}
-{% endfor %}
+{%- endfor %}
     },
     "CONTAINER_FEATURE": {
-{%- for container in ["bgp", "database", "dhcp_relay", "lldp", "pmon", "radv", "restapi", "sflow",
-                      "snmp", "swss", "syncd", "teamd", "telemetry"] %}
+{%- for container in docker_containers_list %}
```
Comment (jleveque) [inline files/build_templates/init_cfg.json.j2]: It appears that Docker containers in the `optional_feature_list` will not get populated here in the `CONTAINER_FEATURE` table.

Also, please note that we are thinking of expanding the concept of a "FEATURE" to include any type of service, whether or not it spawns a container. This way all services can be enabled or disabled from the SONiC CLI. Thus, this logic would need to change.

Comment (stepanblyschak) [reply files/build_templates/init_cfg.json.j2]: @jleveque docker_containers_list is based on $($*_DOCKERS) which includes all optional and non optional docker containers.

Regarding second comment, I can rename SONIC_OPTIONAL_DOCKER_CONTAINERS to SONIC_OPTIONAL_SERVICES, so when you will need to add non-docker service you can append service name to that list and that will work for these services too.

**Comment by liat-grozovik** [comment]
retest vs please

**Comment by liat-grozovik** [comment]
retest vsimage

**Comment by liat-grozovik** [comment]
retest vsimage please

**Comment by liat-grozovik** [comment]
retest vsimage please

**Comment by liat-grozovik** [comment]
retest vsimage please

**Pull request closed without merging**